kedro/io/data_catalog.py

```diff
@@ -520,6 +543,34 @@ def add_feed_dict(self, feed_dict: Dict[str, Any], replace: bool = False) -> Non
 
             self.add(data_set_name, data_set, replace)
 
+    def match_name_against_dataset_factories(self, dataset_input_name: str) -> Optional[AbstractDataSet]:
+        """
+        For a given dataset name, try to match it against the dataset patterns in the catalog.
+        If it's a match, return the dataset instance.
+        """
+        dataset = None
+        # Loop through all dataset patterns and check if the given dataset name has a match.
+        for dataset_name, dataset_config in self.dataset_patterns.items():
```

Comment:
The `dataset_name` here is a `pattern` and the `dataset_config` is a `template`/`factory`, right? We could use that terminology for ease and readability.